### ISSUE TITLE
Unable to return parameters for Basic Block of DRN Network

### ISSUE THREAD

**Issue by sriharsha0806** [sksq96/pytorch-summary#27]
Hi, In DRN [https://github.com/fyu/drn] there is basic block which consists of two series of conv and batchnorm layers for which number of parameters is returned is zero. Can you please look into  this issue? Can you also add parameter indicating the Memory size required for Deployment(forward pass only)

```

       Layer (type)               Output Shape         Param 
================================================================
            Conv2d-1        [-1, 16, 512, 1024]           2,352
       BatchNorm2d-2        [-1, 16, 512, 1024]              32
              ReLU-3        [-1, 16, 512, 1024]               0
            Conv2d-4        [-1, 16, 512, 1024]           2,304
       BatchNorm2d-5        [-1, 16, 512, 1024]              32
              ReLU-6        [-1, 16, 512, 1024]               0
            Conv2d-7         [-1, 32, 256, 512]           4,608
       BatchNorm2d-8         [-1, 32, 256, 512]              64
              ReLU-9         [-1, 32, 256, 512]               0
           Conv2d-10         [-1, 64, 128, 256]          18,432
      BatchNorm2d-11         [-1, 64, 128, 256]             128
             ReLU-12         [-1, 64, 128, 256]               0
           Conv2d-13         [-1, 64, 128, 256]          36,864
      BatchNorm2d-14         [-1, 64, 128, 256]             128
           Conv2d-15         [-1, 64, 128, 256]           2,048
      BatchNorm2d-16         [-1, 64, 128, 256]             128
             ReLU-17         [-1, 64, 128, 256]               0
       BasicBlock-18         [-1, 64, 128, 256]               0
           Conv2d-19         [-1, 64, 128, 256]          36,864
      BatchNorm2d-20         [-1, 64, 128, 256]             128
             ReLU-21         [-1, 64, 128, 256]               0
           Conv2d-22         [-1, 64, 128, 256]          36,864
      BatchNorm2d-23         [-1, 64, 128, 256]             128
             ReLU-24         [-1, 64, 128, 256]               0
       BasicBlock-25         [-1, 64, 128, 256]               0
           Conv2d-26         [-1, 64, 128, 256]          36,864
      BatchNorm2d-27         [-1, 64, 128, 256]             128
             ReLU-28         [-1, 64, 128, 256]               0
           Conv2d-29         [-1, 64, 128, 256]          36,864
      BatchNorm2d-30         [-1, 64, 128, 256]             128
             ReLU-31         [-1, 64, 128, 256]               0
       BasicBlock-32         [-1, 64, 128, 256]               0
           Conv2d-33         [-1, 128, 64, 128]          73,728
      BatchNorm2d-34         [-1, 128, 64, 128]             256
             ReLU-35         [-1, 128, 64, 128]               0
           Conv2d-36         [-1, 128, 64, 128]         147,456
      BatchNorm2d-37         [-1, 128, 64, 128]             256
           Conv2d-38         [-1, 128, 64, 128]           8,192
      BatchNorm2d-39         [-1, 128, 64, 128]             256
             ReLU-40         [-1, 128, 64, 128]               0
       BasicBlock-41         [-1, 128, 64, 128]               0
           Conv2d-42         [-1, 128, 64, 128]         147,456
      BatchNorm2d-43         [-1, 128, 64, 128]             256
             ReLU-44         [-1, 128, 64, 128]               0
           Conv2d-45         [-1, 128, 64, 128]         147,456
      BatchNorm2d-46         [-1, 128, 64, 128]             256
             ReLU-47         [-1, 128, 64, 128]               0
       BasicBlock-48         [-1, 128, 64, 128]               0
           Conv2d-49         [-1, 128, 64, 128]         147,456
      BatchNorm2d-50         [-1, 128, 64, 128]             256
             ReLU-51         [-1, 128, 64, 128]               0
           Conv2d-52         [-1, 128, 64, 128]         147,456
      BatchNorm2d-53         [-1, 128, 64, 128]             256
             ReLU-54         [-1, 128, 64, 128]               0
       BasicBlock-55         [-1, 128, 64, 128]               0
           Conv2d-56         [-1, 128, 64, 128]         147,456
      BatchNorm2d-57         [-1, 128, 64, 128]             256
             ReLU-58         [-1, 128, 64, 128]               0
           Conv2d-59         [-1, 128, 64, 128]         147,456
      BatchNorm2d-60         [-1, 128, 64, 128]             256
             ReLU-61         [-1, 128, 64, 128]               0
       BasicBlock-62         [-1, 128, 64, 128]               0
           Conv2d-63         [-1, 256, 64, 128]         294,912
      BatchNorm2d-64         [-1, 256, 64, 128]             512
             ReLU-65         [-1, 256, 64, 128]               0
           Conv2d-66         [-1, 256, 64, 128]         589,824
      BatchNorm2d-67         [-1, 256, 64, 128]             512
           Conv2d-68         [-1, 256, 64, 128]          32,768
      BatchNorm2d-69         [-1, 256, 64, 128]             512
             ReLU-70         [-1, 256, 64, 128]               0
       BasicBlock-71         [-1, 256, 64, 128]               0
           Conv2d-72         [-1, 256, 64, 128]         589,824
      BatchNorm2d-73         [-1, 256, 64, 128]             512
             ReLU-74         [-1, 256, 64, 128]               0
           Conv2d-75         [-1, 256, 64, 128]         589,824
      BatchNorm2d-76         [-1, 256, 64, 128]             512
             ReLU-77         [-1, 256, 64, 128]               0
       BasicBlock-78         [-1, 256, 64, 128]               0
           Conv2d-79         [-1, 256, 64, 128]         589,824
      BatchNorm2d-80         [-1, 256, 64, 128]             512
             ReLU-81         [-1, 256, 64, 128]               0
           Conv2d-82         [-1, 256, 64, 128]         589,824
      BatchNorm2d-83         [-1, 256, 64, 128]             512
             ReLU-84         [-1, 256, 64, 128]               0
       BasicBlock-85         [-1, 256, 64, 128]               0
           Conv2d-86         [-1, 256, 64, 128]         589,824
      BatchNorm2d-87         [-1, 256, 64, 128]             512
             ReLU-88         [-1, 256, 64, 128]               0
           Conv2d-89         [-1, 256, 64, 128]         589,824
      BatchNorm2d-90         [-1, 256, 64, 128]             512
             ReLU-91         [-1, 256, 64, 128]               0
       BasicBlock-92         [-1, 256, 64, 128]               0
           Conv2d-93         [-1, 256, 64, 128]         589,824
      BatchNorm2d-94         [-1, 256, 64, 128]             512
             ReLU-95         [-1, 256, 64, 128]               0
           Conv2d-96         [-1, 256, 64, 128]         589,824
      BatchNorm2d-97         [-1, 256, 64, 128]             512
             ReLU-98         [-1, 256, 64, 128]               0
       BasicBlock-99         [-1, 256, 64, 128]               0
          Conv2d-100         [-1, 256, 64, 128]         589,824
     BatchNorm2d-101         [-1, 256, 64, 128]             512
            ReLU-102         [-1, 256, 64, 128]               0
          Conv2d-103         [-1, 256, 64, 128]         589,824
     BatchNorm2d-104         [-1, 256, 64, 128]             512
            ReLU-105         [-1, 256, 64, 128]               0
      BasicBlock-106         [-1, 256, 64, 128]               0
          Conv2d-107         [-1, 512, 64, 128]       1,179,648
     BatchNorm2d-108         [-1, 512, 64, 128]           1,024
            ReLU-109         [-1, 512, 64, 128]               0
          Conv2d-110         [-1, 512, 64, 128]       2,359,296
     BatchNorm2d-111         [-1, 512, 64, 128]           1,024
          Conv2d-112         [-1, 512, 64, 128]         131,072
     BatchNorm2d-113         [-1, 512, 64, 128]           1,024
            ReLU-114         [-1, 512, 64, 128]               0
      BasicBlock-115         [-1, 512, 64, 128]               0
          Conv2d-116         [-1, 512, 64, 128]       2,359,296
     BatchNorm2d-117         [-1, 512, 64, 128]           1,024
            ReLU-118         [-1, 512, 64, 128]               0
          Conv2d-119         [-1, 512, 64, 128]       2,359,296
     BatchNorm2d-120         [-1, 512, 64, 128]           1,024
            ReLU-121         [-1, 512, 64, 128]               0
      BasicBlock-122         [-1, 512, 64, 128]               0
          Conv2d-123         [-1, 512, 64, 128]       2,359,296
     BatchNorm2d-124         [-1, 512, 64, 128]           1,024
            ReLU-125         [-1, 512, 64, 128]               0
          Conv2d-126         [-1, 512, 64, 128]       2,359,296
     BatchNorm2d-127         [-1, 512, 64, 128]           1,024
            ReLU-128         [-1, 512, 64, 128]               0
      BasicBlock-129         [-1, 512, 64, 128]               0
          Conv2d-130         [-1, 512, 64, 128]       2,359,296
     BatchNorm2d-131         [-1, 512, 64, 128]           1,024
            ReLU-132         [-1, 512, 64, 128]               0
          Conv2d-133         [-1, 512, 64, 128]       2,359,296
     BatchNorm2d-134         [-1, 512, 64, 128]           1,024
            ReLU-135         [-1, 512, 64, 128]               0
          Conv2d-136          [-1, 19, 64, 128]           9,747
 ConvTranspose2d-137        [-1, 19, 512, 1024]           4,864
      LogSoftmax-138        [-1, 19, 512, 1024]               0
================================================================
```

**Comment by sksq96** [comment]
- Could you provide the exact Network from (DRN)[https://github.com/fyu/drn] you are trying to test? 
- What snippet resulted in the above output and which parameters returned is zero?

**Comment by sriharsha0806** [comment]
* [Code Folder Link](https://drive.google.com/drive/folders/13kb-kSTvXz1VtlB348U3KXKfCWh67a-2?usp=sharing)
* python drnseg.py --arch drn_d_38
*  code snippet is in drnseg.py & code snippet for Basic_Block is in drn.py
```
def main():
	args = parse_args()
	#batch_size = args.batch_size
	#print('calling')
	model = DRNSeg(args.arch, args.classes)
	model = model.cuda().eval()
	images = torch.randn(args.batch_size, 3, 512, 1024)
	images = images.cuda()
	print(summary(model, (3, 512, 1024)))
```
